sugarcrm/client.py

```diff
@@ -0,0 +1,70 @@
+import urllib
+import json
+import hashlib
+import sugarcrm
+from sugarcrm.error import CrmAuthenticationError, CrmResponseError
+
+
+class APIClient:
+    def __init__(self, client=urllib.request, logger=sugarcrm.logger):
+        self.client = client
+        self.session_id = None
+        self.rest_url = sugarcrm.rest_url
+        self.logger = logger
+        encode = hashlib.md5(sugarcrm.password.encode("utf-8"))
```

Comment:
I strongly believe we should move from this line onwards to a `_login` method. The constructor is dealing with too many things.

Reply:
Yeah, I realized during tests that it would be better to separate login. My initial idea was to simplify the code understanding since it makes no sense to call a method not being logged in.